src/monodroid/jni/external-api.cc

```diff
@@ -1,29 +1,69 @@
-#ifdef WINDOWS
-#include <windef.h>
-#include <winbase.h>
-#include <shlobj.h>
-#include <objbase.h>
-#include <knownfolders.h>
-#include <shlwapi.h>
-#endif
-
-#include "globals.hh"
-#include "timing.hh"
+#include <unistd.h>
+#include <mono/utils/mono-publib.h>
+
+#include "monodroid-glue-external.hh"
+
+constexpr int TRUE = 1;
+constexpr int FALSE = 0;
 
 using namespace xamarin::android;
-using namespace xamarin::android::internal;
+
+static MonodroidExternalAPI *external_api = nullptr;
+
+MONO_API bool
+_monodroid_init_external_api (MonodroidExternalAPI *api)
+{
+	if (api == nullptr)
+		return false;
+
+	external_api = api;
+	return true;
+}
+
+/* !DO NOT REMOVE! Used by Mono BCL (System.Net.NetworkInformation.NetworkInterface) */
```

Comment:
Not relevant to this PR, but since I'm curious: how *does* the .NET 5 "hijacking" of `__Internal` mentioned in PR #4757 impact *this* function?  How does `__Internal` *work* in .NET 5, in a way that *we* are even in the picture?  (*Are* we in the picture?)  What happens with `System.Net.NetworkInformation.NetworkInterface`?

Reply:
I have no idea what is the answer here. I haven't investigated the .NET5 `__Internal` stuff yet, not even sure if it's already decide how it will work in the end. However, I *think* `System.Net.NetworkInformation.NetworkInterface` will not use our internal code but some dotnetcore equivalent.